funcx_sdk/funcx/sdk/login_manager/manager.py

```diff
@@ -71,6 +72,16 @@ def run_login_flow(
         *,
         scopes: list[str] | None = None,
     ):
+        # The authorization-via-weblink flow requires stdin; the user must visit
+        # the weblink and enter generated code.
+        if not sys.stdin.isatty() or sys.stdin.closed:
+            # Not technically necessary; the login flow would just die with an EOF
+            # during input(), but adding this message here is much more direct --
+            # handle the non-happy path by letting the user know precisely the issue
+            raise RuntimeError(
+                "Unable to run link-auth login flow: stdin is closed or is not a TTY."
```

Comment:
This has a Globus name: The native app login flow.

Reply:
Ah, had assumed this was the name given the function call below.  Thanks for the education!

Reply:
"native app login flow"